librz/magic/magic-test.c

```diff
@@ -1003,38 +971,34 @@ magic_test_type_regex(RzMagicLine *ml, RzMagicState *ms) {
 				sflag = 1;
 				break;
 			case 'c':
-				flags |= REG_ICASE;
+				flags |= RZ_REGEX_CASELESS;
 				break;
 			default:
 				return (-1);
 			}
 		}
 	}
+	re = rz_regex_new(ml->test_string, RZ_REGEX_EXTENDED | flags, 0, NULL);
+	if (!re) {
+		return -1;
+	}
 
-	if (regcomp(&re, ml->test_string, REG_EXTENDED | REG_NEWLINE | flags) != 0)
-		return (-1);
-	m.rm_so = ms->offset;
-	m.rm_eo = ms->size;
-
-	result = (regexec(&re, ms->base, 1, &m, REG_STARTEND) == 0);
-	if (result == !ml->test_not) {
+	bool res = rz_regex_match(re, ms->base, ms->size, ms->offset, RZ_REGEX_EXTENDED | flags | sflag) != RZ_REGEX_ERROR_NOMATCH;
+	if (res == !ml->test_not) {
 		if (ml->result != NULL) {
-			magic_add_string(ms, ml, ms->base + m.rm_so,
-				m.rm_eo - m.rm_so);
```

Comment:
The old code uses the match offsets from `m`. But the new code uses the offsets from `ms` (bad naming from them tbh).